src/components/cylc/graph/Graph.vue

```diff
@@ -0,0 +1,631 @@
+    // but the vue-cytoscape component offers more than what we need, so we imitate
+    // part of its behaviour here, and ignore the rest and handle the graph and cytoscape
+    // settings ourselves.
+    const element = document.createElement('div')
+    element.setAttribute('class', 'cytoscape')
+    this.$el.appendChild(element)
+    this.cytoscapeInstance = Object.freeze(cytoscape(({
+      container: element
+    })))
+    this.registerExtensions()
+    cytoscape.use(dagre)
+    this.setupEventListeners()
+    this.runLayout(this.cytoscapeInstance)
+    this.setupUndoRedo(this.cytoscapeInstance)
+    this.setupPanzoom(this.cytoscapeInstance)
+    this.setupNavigator(this.cytoscapeInstance)
+    this.setupInteractivity(this.cytoscapeInstance)
+    this.setupHtmlLabel(this.cytoscapeInstance, states)
+    // set the initial zoom-level and pan (position) of the graph
+    // to set only the zoom level, call: this.cytoscapeInstance.zoom(0.5)
+    this.cytoscapeInstance.viewport({
+      zoom: 0.5,
+      pan: {
+        x: 100,
+        y: 100
+      }
+    })
+    this.loading = false
+  },
+
+  beforeDestroy () {
+    // before the component is destroyed, we need to remember to remove the tooltip, otherwise it is still visible
+    // in the next view/route.
+    if (this.tippy !== null) {
+      this.tippy.hide()
+    }
+    this.cytoscapeInstance.panzoom('destroy')
+    document.removeEventListener('keydown', this.keydownEventHandler)
+  },
+
+  computed: {
+    // we map all the workflows here, but we have the workflowName prop to filter and know which one is being displayed
+    ...mapState('workflows', ['workflows'])
+  },
+
+  methods: {
+    /**
+     * @param {[
+     * {
+     *   name: string,
+     *   nodesEdges: {
+     *     edges: [],
+     *     nodes: []
+     *   }
+     * }
+     * ]} workflows - vuex state managed workflows
+     */
+    workflowUpdated_ (workflows) {
+      if (!this.freeze) {
+        this.loading = true
+        const elements = {
+          nodes: [],
+          edges: []
+        }
+        const filtered = workflows.filter((workflow) => workflow.name === this.workflowName)
+        if (filtered.length !== 0) {
+          const workflow = filtered[0]
+          if (Object.hasOwnProperty.call(workflow, 'nodesEdges')) {
+            if (Object.hasOwnProperty.call(workflow.nodesEdges, 'edges')) {
+              workflow.nodesEdges.edges.forEach((edge) => {
+                elements.edges.push({
+                  data: edge
+                })
+              })
+            }
+            if (Object.hasOwnProperty.call(workflow.nodesEdges, 'nodes')) {
+              workflow.nodesEdges.nodes.forEach((node) => {
+                elements.nodes.push({
+                  data: node
+                })
+              })
+            }
+          }
+        }
+        // clear existing data
+        this.graphData.edges.length = 0
+        this.graphData.nodes.length = 0
```

Comment:
Wow I didn't know you could modify an array by messing with its length parameter!

Reply:
Couldn't find a more elegant way to clear lists in JS, but I guess that's normal for JS, I think :grin: